### PR TITLE
Speed up event sampler

### PR DESCRIPTION
**Pull request by fabiopintore**
These PR aims at speeding up the event sampler when a large number of sources has to be simulated, as reported in #5203. The approach is to switch-off the PSF update of the `MapDataset`.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @fabiopintore! Can you please share the performance report before and after this change?

**Comment by fabiopintore** [comment]
I've ran the simulations on my laptop. 
For `gammapy-1.2`, this is the report:
[output_gammapy-1.2.pdf](https://github.com/user-attachments/files/17020188/output_gammapy-1.2.pdf)

while for gammapy-dev:
[output_gammapy-dev.pdf](https://github.com/user-attachments/files/17020191/output_gammapy-dev.pdf)

The simulator seems now significantly faster, according to my test.

**Comment by adonath** [comment]
Thanks @fabiopintore, indeed the absolute runtime is almost a factor of three smaller. Great! Can you add one small unit test and assert that the PSF is indeed none? To avoid that in future some other contributor accidentally reverts the change.

File: gammapy/datasets/simulate.py [hunk 1]
```diff
@@ -247,13 +248,16 @@ def _sample_coord_time(self, npred, temporal_model, gti):
 
         return table
 
-    def sample_sources(self, dataset):
+    def sample_sources(self, dataset, psf_update=None):
```
Comment (AtreyeeS) [inline gammapy/datasets/simulate.py:251]: It might be simpler if `update_psf` is a bool.
If `True`, pass the `evaluator.update` will take `dataset.psf`, else None

**Review by AtreyeeS — CHANGES_REQUESTED** [review]
Thank you @fabiopintore ! This should be very useful for the event sampling! Nice that this did not affect any of the test values... 

I left an inline comment to simplify the API

File: gammapy/datasets/tests/test_simulate.py [hunk 2]
```diff
@@ -354,6 +354,14 @@ def test_mde_sample_sources(dataset, models):
     assert_allclose(events.table["MC_ID"][0], 1, rtol=1e-5)
 
 
+@requires_data()
+def test_mde_sample_sources_psf_update(dataset, models):
+    dataset.models = models
+    sampler = MapDatasetEventSampler(random_state=0)
+    events = sampler.sample_sources(dataset=dataset, psf_update=None)
```
Comment (registerrier) [inline gammapy/datasets/tests/test_simulate.py:361]:
```suggestion
    events = sampler.sample_sources(dataset=dataset, psf_update=False)
```

File: gammapy/datasets/simulate.py [hunk 3]
```diff
@@ -247,27 +248,34 @@ def _sample_coord_time(self, npred, temporal_model, gti):
...
 
-    def sample_sources(self, dataset):
+    def sample_sources(self, dataset, psf_update=False):
         """Sample source model components.
 
         Parameters
         ----------
         dataset : `~gammapy.datasets.MapDataset`
             Map dataset.
+        psf_update : `str` or `~gammapy.datasets.MapDataset`
```
Comment (registerrier) [inline gammapy/datasets/simulate.py:258]:
```suggestion
        psf_update : bool
```

File: gammapy/datasets/simulate.py [hunk 4]
```diff
@@ -247,27 +248,34 @@ def _sample_coord_time(self, npred, temporal_model, gti):
...
+            Parameter to switch-off (on) the update of the PSF
+            in the dataset; available options: `None` or `dataset.psf`.
```
Comment (registerrier) [inline gammapy/datasets/simulate.py:260]:
```suggestion
            in the dataset; default is False.
```

**Review by registerrier — APPROVED** [review]
Thanks @fabiopintore !

**Comment by registerrier** [comment]
CI fail is unrelated. Merging now.

**Comment by adonath** [comment]
Thanks @fabiopintore! I have one additional comment: I would suggest to remove the `psf_update` flag completely in a follow-up PR. I think it has literally has no effect (as the PSF kernel is not used during event sampling), or did I miss something?
